Marlin/Marlin_main.cpp

```diff
@@ -791,9 +792,16 @@ void get_command() {
       char *apos = strchr(command, '*');
       if (npos) {
         gcode_N = strtol(npos + 1, NULL, 10);
-        if (gcode_N != gcode_LastN + 1 && strstr_P(command, PSTR("M110")) == NULL) {
-          gcode_line_error(PSTR(MSG_ERR_LINE_NO));
-          return;
+        if (gcode_N != gcode_LastN + 1) {
+          if (strstr_P(command, PSTR("M110")) == NULL) {
+            gcode_line_error(PSTR(MSG_ERR_LINE_NO));
+            return;
+          }
+          else {
+            npos = strchr(npos, 'N');
+            gcode_N = strtol(npos + 1, NULL, 10);
```

Comment:
Line 794 already does this, does it not? `N123 M110` will set the line number to 123. Your line 801 doesn't do anything, because `npos` is already pointing to the `N` at the start of the line. Your line 802 is redundant because `gcode_LastN = gcode_N` already happens next, on line 815.